cmd/gb-vendor/vendor/repo.go

```diff
@@ -176,8 +185,8 @@ func (g *gitrepo) Checkout(branch, revision string) (WorkingCopy, error) {
 		return nil, err
 	}
 
-	if revision != "" {
-		if err := runOutPath(os.Stderr, dir, "git", "checkout", revision); err != nil {
+	if revision != "" || tag != "" {
+		if err := runOutPath(os.Stderr, dir, "git", "checkout", "-q", oneOf(revision, tag)); err != nil {
```

Comment:
> Nevertheless specifying branch and tag at once should not work at all. 

added a check

> If user specifies branch and tag/revision which is not in that branch it will fail. --single-branch clones only that branch.

I don't know if we can do much to help there, I really don't want to get into the business of wrapping the whole git checkout process -- those flags are really only if you want exactly that specific version of a commit.